packages/node/src/utils/project.ts

```diff
@@ -62,3 +85,159 @@ export function isCustomHandler<K extends string, F>(
 ): handler is SubqlCustomHandler<K, F> {
   return !isBaseHandler(handler);
 }
+
+async function syncRemoteFileToLocal(
```

Comment:
very confuse, why you want reader and content both as parameters?
`content: string` should cover all cases

Reply:
And it's not generic as you tried to make it looks like. It will be easier you just make it `syncIPFSFileToLocal`, and you can use the ipfs hash as filename directly.